tests/Unit/IO/Test_H5.py

```diff
@@ -0,0 +1,89 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from spectre import DataStructures
+from spectre.IO.H5 import IOH5File as  spectre_h5
+import unittest
+import numpy as np
+import os
+
+
+class TestIOH5File(unittest.TestCase):
+    # Test Fixtures
+    def setUp(self):
+        self.file_name = "./pythontest.h5"
+        self.data_1 = [1.0, 3.5]
+        self.data_1_array = np.array(self.data_1)
+        self.data_2 = [3.0, 10.3]
+        self.data_2_array = np.array(self.data_2)
+        self.cd = os.getcwd()
+        if "pythontest.h5" in os.listdir(self.cd):
+            os.remove(self.file_name)
+
+    # Test whether an H5 file is created correctly, delete a file with the name
+    # 'pythontest.h5' if it already exists.
+    def test_name(self):
+        file_spec = spectre_h5.H5File(self.file_name, 1)
+        self.assertEqual(self.file_name, file_spec.name())
+        file_spec.close()
+        os.remove(self.file_name)
+
+    # Test whether a dat file can be added correctly
+    def test_insert_dat(self):
+        file_spec = spectre_h5.H5File(self.file_name, 1)
+        file_spec.insert_dat("/element_data", ["Time", "Value"], 0)
+        datfile = file_spec.get_dat("/element_data")
+        self.assertEqual(datfile.get_version(), 0)
+        file_spec.close()
+        os.remove(self.file_name)
+
+    # Test whether data can be added to the dat file correctly
+    def test_append(self):
+        file_spec = spectre_h5.H5File(self.file_name, 1)
+        file_spec.insert_dat("/element_data", ["Time", "Value"], 0)
+        datfile = file_spec.get_dat("/element_data")
+        datfile.append(self.data_1)
+        outdata_array = datfile.get_data()
+        assert(outdata_array == self.data_1_array).all()
```

Comment:
Does this test run? I think better `import numpy.testing as npt` and `npt.assert_array_equal(a, b)`